Assets/Scripts/Edges/LineFactory.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System.Collections.Generic;
+using Assets.Scripts.Common.Utils;
+using UnityEngine;
+
+namespace Assets.Scripts.Edges
+{
+  public class LineFactory : MonoBehaviour
+  {
+    public BezierCurveFactory bezierCurveFactory;
+    public bool debug;
+    public DebugSphereFactory debugSphereFactory;
+
+    /// <summary>
+    ///   Returns a list of points that make up a curved, rotated edge when <paramref name="numberOfLines" /> > 1
+    ///   and a straight, two points line when it is == 1
+    /// </summary>
+    public IList<IList<Vector3>> GetLinePositionsFor(Vector3 startingPoint, Vector3 endingPoint, int numberOfLines)
+    {
+      switch (numberOfLines)
+      {
+        case 0:
+          return new List<IList<Vector3>>();
+        case 1:
+          return GetEdgePointsForSingleEdge(startingPoint, endingPoint);
+        default:
+          return GetEdgePointsForMultipleEdges(startingPoint, endingPoint, numberOfLines);
+      }
+    }
+
+    IList<IList<Vector3>> GetEdgePointsForSingleEdge(Vector3 startingPoint, Vector3 endingPoint) =>
+      new List<IList<Vector3>> {new List<Vector3> {startingPoint, endingPoint}};
```

Comment:
Oof.
This needs a comment

Reply:
xD I know. It's quite complicated though. Edges always have to be considered in groups, because position of one edge is dependent on others. I know it's not pretty but I couldn't figure out anything better